reports/tests.py

```diff
@@ -0,0 +1,508 @@
+
+        url = reverse(self.url, kwargs={
+            'organization': self.organization.github_handle,
+            'year': '2018',
+            'month': invalid_month
+        })
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 400)
+
+    def test_report_view_no_organization(self):
+        """
+        No organization means no report, simply we're returning a 404 response.
+        """
+        wrong_org = {
+            'organization': 'bad_org',
+            'year': '2018',
+            'month': '10'
+        }
+
+        self.client.login(username=self.username, password=self.password)
+
+        url = reverse(self.url, kwargs=wrong_org)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 404)
+
+    def test_report_view_no_forks(self):
+        """
+        If an organization doesn't have a any foks then the total charges
+        must be 0
+        """
+        self.client.login(username=self.username, password=self.password)
+
+        url = reverse(self.url, kwargs={
+            'organization': self.organization.github_handle,
+            'year': '2018',
+            'month': '10'
+        })
+        response = self.client.get(url)
+        self.assertContains(response, 'TOTAL including all forks: 0 €')
+
+    @mock.patch("reports.views.generate_charges")
+    def test_report_view_good_behavior(self, generate_charges_mock):
+        """
+        Good behavior here should return a 200 response
+        """
+        self.client.login(username=self.username, password=self.password)
+        url = reverse(self.url, kwargs={
+            'organization': self.organization.github_handle,
+            'year': '2018',
+            'month': '10'
+        })
+
+        generate_charges_mock.return_value = ({}, 10)
+        response = self.client.get(url)
+        self.assertContains(response, 'TOTAL including all forks: 10 €')
+
+
+@ddt.ddt
+class ReportsHelpersTestCase(TestCase):
+    """
+    Tests for reports helpers functions
+    """
+
+    def setUp(self):
+        self.appserver = make_test_appserver()
+        self.organization = OrganizationFactory.create(
+            name='Organization Name',
+            github_handle='organization_github_handle'
+        )
+
+    @ddt.data(
+        (2000, 1, 1, 0, 0, 0, 0),
+        (2018, 5, 12, 10, 30, 20, 40),
+        (2222, 12, 31, 23, 59, 59, 1000),
+    )
+    @ddt.unpack
+    def test_normalize_date(self, year, month, day, hour, minute, second, microsecond):  # pylint: disable=too-many-arguments
+        """
+        Will test the default case and the edge cases to truncate the date.
+        """
+        date = datetime(year, month, day=day, hour=hour, minute=minute, second=second, microsecond=microsecond)
+
+        new_date = _normalize_date(date)
+        expected_date = datetime(date.year, date.month, 1)  # Should keep the same year and month
```

Comment:
This is running some Python code to get the expected value. In a test it would be better to explicitly say which value is expected, for instance passing it in the ddt parameters.
But this function is simple and we don't have much time, so we can leave it like this.